### PR TITLE
Fixed #26215 -- Fixed RangeField.value_to_string() serialization for null range boundaries.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/contrib/postgres/fields/ranges.py [hunk 1]
```diff
@@ -51,8 +51,11 @@ def value_to_string(self, obj):
         base_field = self.base_field
         result = {"bounds": value._bounds}
         for end in ('lower', 'upper'):
-            obj = AttributeSetter(base_field.attname, getattr(value, end))
-            result[end] = base_field.value_to_string(obj)
+            if getattr(value, end) is None:
```
Comment (timgraham) [inline django/contrib/postgres/fields/ranges.py]: maybe it's worth creating a variable to avoid a possible `getattr` twice?

**Comment by timgraham** [comment]
Sounds good to backport to 1.8/1.9. Might be worth checking the other serialization formats too. We have the infrastructure in `tests/serializers/test_data.py` for creating other tests as needed. I'd be fine if you test this manually for now and create a new ticket to improve the `postgres_tests` if it's warranted.

File: django/contrib/postgres/fields/ranges.py (hunk 1 above)
Comment (claudep) [inline django/contrib/postgres/fields/ranges.py]: If you think the overhead of a new variable is lower than the overhead of two getattr, then fine. But I'm not sure about it.

Comment (timgraham) [inline django/contrib/postgres/fields/ranges.py]: Not sure either, I was thinking mostly about readability.

**Comment by timgraham** [comment]
I had to restart Jenkins so this build got lost - I'll retrigger. Feel free to use the PR title as a release note.

**Comment by timgraham** [comment]
buildbot, test this please.

File: tests/postgres_tests/test_hstore.py [hunk 2]
```diff
@@ -165,16 +165,16 @@ def test_usage_in_subquery(self):
 
 
 class TestSerialization(PostgreSQLTestCase):
-    test_data = '[{"fields": {"field": "{\\"a\\": \\"b\\"}"}, "model": "postgres_tests.hstoremodel", "pk": null}]'
+    test_data = '[{"fields": {"field": "{\\"a\\": \\"b\\", \\"c\\": null}"}, "model": "postgres_tests.hstoremodel", "pk": null}]'
```
Comment (timgraham) [inline tests/postgres_tests/test_hstore.py]: The test failure is a non-deterministic dict ordering issue within "field".

**Pull request merged**